src/TrackingOptOut.php

```diff
@@ -9,60 +9,84 @@
 
 class TrackingOptOut {
 
-  /**
-   * Default Google Analytics opt out script.
-   *
-   * @var string
-   */
-  private static $opt_out_script = <<<CODE
-<script>
-// Disable tracking if the opt-out cookie exists.
-for (var i = 0; i < window.gaIdList.length; i++) {
-  var disableStr = 'ga-disable-' + window.gaIdList[i];
-  if (document.cookie.indexOf(disableStr + '=true') > -1) {
-    window[disableStr] = true;
-  }
-}
-/**
- * Sets cookies to indicate Google Analytics tracking should be disabled.
- */
-function gaOptout() {
-  var disableStr;
-  for (var i = 0; i < window.gaIdList.length; i++) {
-    disableStr = 'ga-disable-' + window.gaIdList[i];
-    document.cookie = disableStr + '=true; expires=Thu, 31 Dec 2099 23:59:59 UTC; path=/';
-    window[disableStr] = true;
-  }
-}
-</script>
-
-CODE;
-
   /**
    * @implements admin_init
    */
   public static function admin_init() {
-    // Register opt-out related settings.
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_ids');
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_script');
-    register_setting(Plugin::PREFIX . '-settings', 'core_standards_optout_active');
-
+    static::addSettingsFields();
     add_action('admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts');
   }
 
   /**
    * @implements init
    */
   public static function init() {
-    // Injects the Google Analytics IDs and the opt-out script in the page head
-    // section before any tracking script.
-    add_action('wp_head', __CLASS__ . '::wp_head', 5);
 
     // Register the shortcode to call the opt-out cookie creation script.
     add_shortcode('ga-optout-link', __CLASS__ . '::registerOptOutShortcode');
 
     // Enqueue frontend scripts.
-    add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+    if (get_option('core_standards_optout_active')) {
+      add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+      // Injects the Google Analytics IDs and the opt-out script in the page head
+      // section before any tracking script.
+      add_action('wp_head', __CLASS__ . '::wp_head', 5);
+      add_action('wp_enqueue_scripts', __CLASS__ . '::enqueueScriptsFirstHead', 1);
+    }
+  }
+
+  /**
+   * Adds Opt-Out section settings fields.
+   */
+  public static function addSettingsFields() {
+    add_settings_section(
+      Plugin::PREFIX . '-optout',
+      __('WordPress Core Standards Settings', Plugin::L10N),
+      NULL,
+      'general'
+    );
+
+    add_settings_field(
+      'core_standards_optout_ids',
+      __('Google Analytics IDs', Plugin::L10N),
+      __CLASS__ . '::optOutGoogleIdsFieldRender',
+      'general',
+      Plugin::PREFIX . '-optout'
+    );
+
+    add_settings_field(
+      'core_standards_optout_active',
```

Comment:
This separate activation setting looks unnecessary.  The feature should automatically be toggled depending on whether GA IDs were configured — there is no point in enabling the feature without GA IDs.

Reply:
@sun I want to keep the separate activation control for two reasons:
1. I don't like having a single interface element (GA IDs text box) doing two different functions; that's not good UX, IMO.
2. This way opt-out can be disabled without having to wipe the GA IDs list. No need to remember/searching for the GA IDs if user deactivated opt-aut and wants to reactivate it later on.

What I will do is adding a condition to disable opt-out if the GA IDs is empty.